sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+// TODO: remove this
+#pragma warning(disable : 4239)
+
+#include "http/http.hpp"
+#include "common/xml_wrapper.hpp"
+
+#include <map>
+#include <string>
+#include <stdio.h>
+#include <time.h>
+#include <stdint.h>
+#include <ctime>
+
+namespace Azure
+{
+namespace Storage
+{
+namespace Blobs
+{
+class BlobRestClient
+{
+  const static std::string k_HEADER_METADATA_PREFIX;
+  const static std::string k_HEADER_DATE;
+  const static std::string k_HEADER_X_MS_VERSION;
+  const static std::string k_HEADER_AUTHORIZATION;
+  const static std::string k_HEADER_CLIENT_REQUEST_ID;
+  const static std::string k_HEADER_ETAG;
+  const static std::string k_HEADER_LAST_MODIFIED;
+  const static std::string k_HEADER_X_MS_CLIENT_REQUEST_ID;
+  const static std::string k_HEADER_X_MS_REQUEST_ID;
+  const static std::string k_HEADER_CONTENT_MD5;
+  const static std::string k_HEADER_X_MS_CONTENT_CRC64;
+  const static std::string k_RESTYPE;
+  const static std::string k_CONTAINER;
+  const static std::string k_QUERY_COMP;
+  const static std::string k_QUERY_BLOCK_LIST;
+  const static std::string k_QUERY_BLOCK;
+  const static std::string k_BLOB;
+  const static std::string k_BLOCK_ID;
+
+public:
+  struct RequestOptions
+  {
+    std::string Version = "2019-02-02";
+    std::string ClientRequestID = std::string();
+    std::string Date = std::string();
+  };
+
+  struct BodiedRequestOptions : public RequestOptions
+  {
+    azure::core::http::BodyBuffer* BodyBuffer = nullptr;
+    azure::core::http::BodyStream* BodyStream = nullptr;
+  };
+
+  struct ResponseInfo
+  {
+    std::string RequestId = std::string();
+    std::string Date = std::string();
+    std::string Version = std::string();
+    std::string ClientRequestID = std::string();
+  };
+
+  class Container
+  {
+    const static std::string k_HEADER_MS_BLOB_PUBLIC_ACCESS;
+
+  public:
+    enum class PublicAccessType
+    {
+      Container = 0,
+      Blob,
+      Anonymous
```

Comment:
I think this should be `Private`.

Reply:
Public access type should be specifiable by customer/convenience layer.

Reply:
according to the REST API reference https://docs.microsoft.com/en-us/rest/api/storageservices/create-container, if neither `container` or `blob` is specified, then the container is private. I don't think `Anonymous` here means private.